vals_test.go

```diff
@@ -116,3 +116,37 @@ kind: Secret
 
 	require.Equal(t, expected, buf.String())
 }
+
+func TestEvalIntegerBool(t *testing.T) {
+	var yamlDocs = `
+---
+Integer: 1
+ResultInteger: ref+file://secrets.yaml#/Integer
+Bool: true
+ResultBool: ref+file://secrets.yaml#/Bool
+`
+	var expected = `Bool: true
+Integer: 1
+ResultBool: "true"
+ResultInteger: "1"
```

Comment:
@artiodev Thanks a bunch for the pull request!
This looks well aligned with vals' design, where it wants any fetched values to be rendered as strings, so that the values before(`ref+...`) and after the vals rendering are both strings. It's great in that regard.

But, is this what you originally wanted in #190?

Here, I was wondering if you wanted:
```suggestion
ResultBool: true
ResultInteger: 1
```

rather than the string representations of the original values.

If necessary, we could support both options, perhaps via feature toggling envvars and or ref URL params. Just wanted to confirm your goal and intention to start!

Thanks in advance for your cooperation 🙏